x-pack/plugins/security_solution/server/lib/telemetry/queries.ts

```diff
@@ -0,0 +1,274 @@
+import { ElasticsearchClient, Logger, SavedObjectsClient } from 'src/core/server';
```

Comment:
The idea, IMO, is that this module is going to capture ES queries inside the telemetry library. Let me know if this is a bad idea -- generally seemed good to abstract this portion from the actual transmission.

Reply:
I think this is a good idea in theory, but the execution is problematic and doesn't set us up to adequately test this.
A more effective way of writing this code would be to invert control and pass the `TelemetryQuerier` as a dependency to the `TelemetrySender` class.

```ts
const telemetryQuerier = new TelemetryQuerier(...deps)
const telemetrySender = new TelemetrySender(...deps, telemetryQuerier)
```

By doing this the sender no longer needs to reference an SO Client, ES Client, Fleet Interface, etc and will get us closer to a Single Responsibility telemetry sender. 

```ts
const telemetryQuerier = new TelemetryQuerier(...deps)
const telemetrySender = new TelemetrySender(...deps)
const telemetryOrchestrator = new TelemetryOrchestrator(telemetryQuerier, telemetrySender, ...deps)
```

I also consider the term `Querier` is overloaded here - `Receiver` would be a better noun to describe the interface interactions.
Let's pair on this bit.